pybop/models/lithium_ion/echem_base.py

```diff
@@ -31,16 +31,24 @@ def _check_params(
         """
         parameter_set = parameter_set or self._parameter_set
 
-        electrode_params = [
-            (
-                "Negative electrode active material volume fraction",
-                "Negative electrode porosity",
-            ),
-            (
-                "Positive electrode active material volume fraction",
-                "Positive electrode porosity",
-            ),
-        ]
+        if self.pybamm_model.options["working electrode"] == "positive":
```

Comment:
I am not very happy with this hack, but better solutions would require significant refactoring that should be discussed first.